fhir-server/src/main/java/au/csiro/pathling/Configuration.java

```diff
@@ -296,7 +299,20 @@
      * Controls the maximum number of cache entries held in memory.
      */
     @NotNull
-    private long maxEntries;
+    @Min(0)
+    private long aggregateRequestCacheSize;
```

Comment:
There is a bit of inconsistency of how the primitive types are represented here, in particular in case of @NotNull fields.
E.g. : 
```
@NotNull
    @Min(0)
    private Integer socketTimeout;
```

vs 

```
    @NotNull
    @Min(0)
    private long searchPageCacheSize;
```
I think in case of @NonNull fields it make sense to represent them as primitive types.

Reply:
I can't work out if you still need `@NotNull` if the type is primitive.

I think I'll just standardise on primitive types and keep the `@NotNull` annotations...

Reply:
Actually, that doesn't seem to work.

If I change `socketTimeout` to an `int`, then remove it from `application.yml`, it seems to load without error but assign `0` as the value. This is not the behaviour that we want, we want it to throw an error in this case.

I think this might have been the reason that I used the boxed primitive for that value in the first place.

Reply:
I can confirm that it all works as desired when using the boxed primitives.

So maybe the action here is to standardise on the boxed primitives, including those `long` values that I have added as part of this PR.

Reply:
See 5ad8d4094a18c8c18f84aec8fe3fa139338482fc.